### DISCUSSION TITLE
Matrix Multiplication - Matrix dot Matrix

### DISCUSSION THREAD

**Discussion by D1Asif** [mrdbourke/pytorch-deep-learning#574] [Q&A] [answered]
From maths, we know that to multiply a matrix with another matrix the order of the matrix has to be like this:
1st matrix: m x n
2nd matrix: n x p
that is, 1st matrix's number of column has to be equal to the number of rows of the second matrix. But in the video, we can see that [1, 2, 3] dot [1, 2, 3] is 14. How is it possible?

**Comment by Perian-Yan** [comment]
Hi,
do you mean [`pytorch.dot()`](https://pytorch.org/docs/stable/generated/torch.dot.html)? If so, it's used for computing the inner product of two 1D vectors. 

There are two types of multiplication: one is matrix multiplication, as you said. We can use [`torch.matmul(a,b)`](https://pytorch.org/docs/stable/generated/torch.mul.html) or `@`. The other one is element-wise multiplication and we use `*` or [`torch.mul(a,b)`](https://pytorch.org/docs/stable/generated/torch.mul.html)

Hope this helps.

**Reply by D1Asif** [reply]
I was asking why the `torch.matmul(a, b)` = 14 if a = b = [1, 2, 3]. Shouldn't it be an error? Since the inner dimensions don't match?

**Comment by Perian-Yan** [comment, marked as answer]
In the [torch.matmul(a,b)](https://pytorch.org/docs/stable/generated/torch.matmul.html) document: If both tensors are 1-dimensional, the dot product (scalar) is returned.

**Reply by D1Asif** [reply]
looks like the sum of all the elements of scalar product is returned.
